### ISSUE TITLE
Register user

### ISSUE THREAD

**Issue by Prior99** [Rantanen/node-mumble#104]
Is it possible, using `node-mumble` to register another user if the user which `node-mumble` authenticated with has the necessary permissions to do so (e.g. connected as SuperUser)? I could neither find a method for this nor something in mumbles Protocol buffer.

**Comment by Rantanen** [comment]
Hum. It should be possible since the Mumble client does this already and the only thing it has is the protocol buffer interfaces.

**Comment by Rantanen** [comment]
https://github.com/mumble-voip/mumble/blob/master/src/murmur/Messages.cpp#L730-L749

Seems to be just a case of sending an updated user info with the user_id filled. Don't think the actual ID is used so as long as it's something else than null it should work I'd guess. Can you try this? I would welcome some sort of pull request to have a `.register()` method on the user objects if you get this working.

**Comment by Prior99** [comment]
Sure. I will take a look at it.

**Comment by Prior99** [comment]
See #105. The id assigned to the user needs to be calculated by the client updating the user information, hence I implemented another utility in that PR: `getRegisteredUsers`. It can only be called if the ACL allow the user to register other users. I'd be happy if you could review and publish this.

**Comment by Prior99** [comment]
I also found it weird, but when trying to simply call it with `1` for two users it would succeed only the first time on a clean server. Everytime I re-used an already assigned id I got a `PermissionError`. Fixing it to assigning truely unique ids worked fine, though.

By the way, I might be mistaken but I think the ACL for registering also allows to list registered users.

**Comment by Rantanen** [comment]
Yeah, ACL for registering (as opposed to self registering) is needed to retrieve the list of users. And this ACL needs to be defined on the root channel:

https://github.com/mumble-voip/mumble/blob/master/src/murmur/Messages.cpp#L1511-L1516

**Comment by Prior99** [comment]
I think this can be closed after #105 was merged, what do you think @Rantanen?
